src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -131,7 +131,7 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   protected def findEntry(key: A): Entry =
     findEntry0(key, index(elemHashCode(key)))
 
-  private[this] def findEntry0(key: A, h: Int): Entry = {
+  protected[this] final def findEntry0(key: A, h: Int): Entry = {
```

Comment:
It would also be see whether performance of other methods in `mutable.HashMap` could be improved by using copy/pasted implementations of `findEntry` etc rather than those inherited from the `HashTable` trait. Obviously this isn't an ideal solution, but `HashMap` is so widely used, and these methods are so tiny that failure to JIT inline can have a real impact of observable performance, that I'd be open to this solution for 2.12.1.